packages/mobx/src/core/reaction.ts

```diff
@@ -195,10 +195,23 @@ export class Reaction implements IDerivation, IReactionPublic {
         }
     }
 
-    getDisposer_(): IReactionDisposer {
+    getDisposer_(abortSignal?: GenericAbortSignal): IReactionDisposer {
         const r = this.dispose.bind(this) as IReactionDisposer
```

Comment:
Since we already have to create new function here I think we could merge these cases?
```ts
const dispose = () => {
  this.dispose();
  abortSignal?.removeEventListener?.("abort", dispose);
}
abortSignal?.addEventListener?.("abort", dispose)
dispose[$mobx] = this
return dispose;
```